org.uqbar.project.wollok.typeSystem/src/org/uqbar/project/wollok/typesystem/constraints/ConstraintGenerator.xtend

```diff
@@ -160,20 +161,24 @@ class ConstraintGenerator {
 	}
 
 	def dispatch void generateVariables(WListLiteral it) {
-		val listType = newCollection(classType(LIST))
```

Comment:
Method new collection is no longer necessary, collection are just regular type variables. All (non void?) type variables accept type parameters (i.e. they are Generic).